client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
```

Comment:
also I'm kind of curious on your name `observer`. when I created something like this, my brainstorming led me to `handler` instead. the thinking was that since these functions get called by the middleware on certain events, they were thus event handlers.

I wouldn't normally care about the naming here, but _observer_ seems to carry with it some implications about _observables_ in general, as in I expect these to be _watching_ for something when in fact they don't do anything actively - all passively being called by something else. they're not even inserting a hook somewhere, just living in a data structure.